File: 00_connection_examples/example_query_simple.py
# INSTALL psycopg2 ON YOUR PYTHON DISTRIBUTION IF NOT ALREADY INSTALLED

# Import packages
import os
from package_DataProcessing import connect_database_postgresql
from package_DataProcessing import query_to_dataframe

# Set file paths
authentication_path = 'N:/ACCS_Work/Administrative/Credentials/akveg_private_read' # UPDATE FILE PATH

# Identify files
authentication_file = os.path.join(authentication_path, 'authentication_akveg_private.csv')

# Create a connection to the AKVEG Database
database_connection = connect_database_postgresql(authentication_file)

# Query the database and return result as dataframe
query = 'SELECT * FROM site'
query_table = query_to_dataframe(database_connection, query)

# Print dataframe
print(query_table)

File: 00_connection_examples/example_query_simple.R
# Import required libraries
library(RPostgres)
library(tibble)

# Set file paths
repository_path = 'C:/Users/timmn/Documents/Repositories/akveg-database' # UPDATE FILE PATH
authentication_path = 'N:/ACCS_Work/Administrative/Credentials/akveg_private_read' # UPDATE FILE PATH

# Identify files
connection_script = paste(repository_path, 'package_DataProcessing', 'connect_database_postgresql.R',
                          sep = '/')
authentication_file = paste(authentication_path, 'authentication_akveg_private.csv',
                            sep = '/')

# Import database connection function
source(connection_script)

# Create a connection to the AKVEG Database
database_connection = connect_database_postgresql(authentication_file)

# Query the database and return result as tibble
query = 'SELECT * FROM site'
query_table = as_tibble(dbGetQuery(database_connection, query))